code/obj/flock/structure/flock_structure_ghost.dm

```diff
@@ -46,3 +46,22 @@
 	if(src.building)
 		new building(get_turf(src), src.flock)
 	qdel(src)
+
+/obj/flock_structure/ghost/proc/cancelBuild()
+	if (currentmats > 0)
+		var/obj/item/flockcache/cache = new(get_turf(src))
+		cache.resources = currentmats
+	flock_speak(null, "Tealprint derealizing", flock)
+	playsound(src, 'sound/misc/flockmind/flockdrone_door_deny.ogg', 50, 1)
+	qdel(src)
+
+/obj/flock_structure/ghost/verb/cancel()
```

Comment:
Verbs can be seen from anywhere and a bit cumbersome to use. Maybe just interacting with a ghost building can open some sort of UI or similar that has a button to cancel?

Reply:
Yeah, agree. We're looking to remove verbs entirely eventually.

Reply:
even just opening a tgui alert yes/no

Reply:
alright, I'll replace it with a tgui alert for now, and I'll probably add a cancel button to the flockpanel as well in future